### DISCUSSION TITLE
Add tie range (second shortest path length) algorithm

### DISCUSSION THREAD

**Discussion by KatyYuu** [networkx/networkx#7068] [Ideas]
Mark Granovetter's renowned paper, "The Strength of Weak Ties,"[1] presents a unique bridge distance metric termed the "local bridge of degree n." This metric represents the second shortest path length between two directly connected nodes, with the shortest path being the direct edge itself. While this measure has been referenced in various studies[2], it hasn't been incorporated into Networkx yet.

I would kindly like to ask the maintainers of NetworkX to add this algorithm as a new feature.

> [1] Granovetter, Mark S. “The Strength of Weak Ties.” American Journal of Sociology, vol. 78, no. 6, 1973, pp. 1360–80. JSTOR, http://www.jstor.org/stable/2776392. Accessed 27 Oct. 2023.`

> [2] Patrick S. Park et al. ,The strength of long-range ties in population-scale social networks.Science362,1410-1413(2018).DOI:[10.1126/science.aau9735](https://doi.org/10.1126/science.aau9735)

**Comment by henryxu1997** [comment]
Hi Katy, this is a great suggestion and this is something I am interested in. I would love to contribute and implement this feature, so I am writing a comment here to claim this feature.
